File: app/src/main/java/com/isp/app/ui/programs/ProgramsViewModel.java
package com.isp.app.ui.programs;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.isp.app.data.models.Program;
import com.isp.app.network.ApiService;
import java.util.List;

public class ProgramsViewModel extends ViewModel {
    private final ApiService apiService;
    private final MutableLiveData<List<Program>> programs = new MutableLiveData<>();
    private final MutableLiveData<Boolean> loading = new MutableLiveData<>();
    private final MutableLiveData<String> error = new MutableLiveData<>();

    public ProgramsViewModel(ApiService apiService) {
        this.apiService = apiService;
    }

    public LiveData<List<Program>> getPrograms() {
        return programs;
    }

    public LiveData<Boolean> getLoading() {
        return loading;
    }

    public LiveData<String> getError() {
        return error;
    }

    public void loadPrograms() {
        new Thread(() -> {
            try {
                loading.postValue(true);
                error.postValue(null);
                programs.postValue(apiService.getPrograms());
            } catch (Exception e) {
                error.postValue(e.getMessage());
            } finally {
                loading.postValue(false);
            }
        }).start();
    }
} 

File: app/src/main/java/com/college/app/ui/programs/ProgramsViewModel.kt
package com.college.app.ui.programs

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.college.app.data.models.Program
import com.college.app.network.ApiService
import kotlinx.coroutines.launch
import javax.inject.Inject

class ProgramsViewModel @Inject constructor(
    private val apiService: ApiService
) : ViewModel() {

    private val _programs = MutableLiveData<List<Program>>()
    val programs: LiveData<List<Program>> = _programs

    private val _loading = MutableLiveData<Boolean>()
    val loading: LiveData<Boolean> = _loading

    private val _error = MutableLiveData<String?>()
    val error: LiveData<String?> = _error

    fun loadPrograms() {
        viewModelScope.launch {
            try {
                _loading.value = true
                _error.value = null
                _programs.value = apiService.getPrograms()
            } catch (e: Exception) {
                _error.value = e.message
            } finally {
                _loading.value = false
            }
        }
    }
} 